charts/redpanda/values.yaml

```diff
@@ -421,18 +421,17 @@ resources:
       max: 2.5Gi
     #
     # This optional `redpanda` object allows you to specify the memory size for both the Redpanda
-    # process and the underlying reserved memory used by Seastar.
+    # process and the Seastar subsystem.
     # This section is omitted by default, and memory sizes are calculated automatically
     # based on container memory.
     # Uncommenting this section and setting memory and reserveMemory values will disable
     # automatic calculation.
     #
-    # If you are setting the following values manually, keep in mind the following guidelines.
-    # Getting this wrong may lead to performance issues, instability, and loss of data:
-    # The amount of memory to allocate to a container is determined by the sum of three values:
-    # 1. Redpanda (at least 2Gi per core, ~80% of the container's total memory)
-    # 2. Seastar subsystem (200Mi * 0.2% of the container's total memory, 200Mi < x < 1Gi)
-    # 3. Other container processes (whatever small amount remains)
+    # If you are setting these values manually, follow these guidelines carefully. Incorrect settings can lead to performance degradation, instability, or even data loss. The total memory allocated to a container is determined as the sum of the following two areas:
+    #
+    #- Redpanda (including Seastar): Defined by the `--memory` parameter. Includes the memory used by the Redpanda process and the reserved memory allocated for Seastar. A minimum of 2Gi per core is required, and this value typically accounts for ~80% of the container’s total memory. For production, allocate at least 8Gi.
+    #
+    # - Operating system (OS): Defined by the `--reserve-memory` parameter. Represents the memory available for the operating system and other processes within the container.
```

Comment:
As luck would have it, I was just poking around at these values myself. As I understand it, these values are passed directly through to seastar. Meaning that they should have the same meaning as they do in scylladb.

According to [their docs](https://opensource.docs.scylladb.com/stable/kb/memory-usage), these flags should be considered mutually exclusive:
> To limit the memory usage you can start scylla with --memory parameter. Alternatively, you can specify the amount of memory ScyllaDB should leave to the OS with --reserve-memory parameter. Keep in mind that the amount of memory left to the operating system needs to suffice external scylla modules.


See also this issue and comment from our very own Travis Downs: https://github.com/scylladb/seastar/issues/375#issuecomment-1081104600


Assuming that seastar correctly detects the memory limits imposed by the cgroups configured by Kubernetes, it seems like we should be discouraging the use of these flags when running in Kubernetes.

Reply:
Added a warning to discourage their use.

Reply:
I'm not sure the description is correct. Redpanda does not really "reserve" memory per se. It only allocates memory for its heap, specified by `--memory`. The when `--memory` is used the `--reserved` value is basically just a check that the `--memory` amount is not too big of a fraction of total available memory: in that case it will produce an error and fail to start.

If you are _not_ using `--memory` then `--reserved` is much more important: it basically sets the heap size based on the total available memory, minus the amount passed as `--reserved`. Note that if you don't pass reserved, seastar is calculating a default reserved amount anyway (the greater of 1.5 GB or 7% of memory).

I think the --reserved option is most important when you do not have somethign like a helm chart to do the exact `--memory` calculation for you, in that case it's a way of letting seastar auto-detect how much memory to use. When the helm chart it mostly just gets in the way since we are already doing a calculation to set --memory based on total memory.